code/modules/reagents/Chemistry-Reagents/Consumable-Reagents/Food-Reagents.dm

```diff
@@ -52,6 +52,25 @@ datum/reagent/consumable/sugar
 	reagent_state = SOLID
 	color = "#FFFFFF" // rgb: 255, 255, 255
 	nutriment_factor = 10 * REAGENTS_METABOLISM
+	metabolization_rate = 2 * REAGENTS_METABOLISM
+	overdose_threshold = 200 // Hyperglycaemic shock
+
+datum/reagent/consumable/sugar/on_mob_life(var/mob/living/M as mob)
+	if(M.satiety > -200)
+		M.satiety -= 20 //eating sugar makes you more hungry over time by making your hunger drop faster.
+	M.nutrition += max(((NUTRITION_LEVEL_FED + 50) - M.nutrition )/100, 0) * nutriment_factor // sugar doesn't help your hunger if your stomach is nearly full
+	holder.remove_reagent(src.id, metabolization_rate)
```

Comment:
This could of been a error from having a older version of the code base but I dont know someone should fix it quick.